src/controller/artifact/processor/model/parser/base.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package parser
+
+import (
+	"context"
+	"fmt"
+
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+
+	"github.com/goharbor/harbor/src/pkg/artifact"
+	"github.com/goharbor/harbor/src/pkg/registry"
+)
+
+const (
+	// contentTypeTextPlain is the content type of text/plain.
+	contentTypeTextPlain = "text/plain; charset=utf-8"
+	// contentTypeTextMarkdown is the content type of text/markdown.
+	contentTypeMarkdown = "text/markdown; charset=utf-8"
+	// contentTypeJSON is the content type of application/json.
+	contentTypeJSON = "application/json; charset=utf-8"
+)
+
+// NewBase creates a new base parser.
+func NewBase(cli registry.Client) *base {
+	return &base{
+		regCli: cli,
+	}
+}
+
+// base provides a default implementation for other parsers to build upon.
+type base struct {
+	regCli registry.Client
+}
+
+// Parse is the common implementation for parsing layer.
+func (b *base) Parse(_ context.Context, artifact *artifact.Artifact, layer *ocispec.Descriptor) (string, []byte, error) {
+	if artifact == nil || layer == nil {
```

Comment:
please add the blob size checking, if the blob is larger than a specific size, then UI shows unable to show the file content.

Reply:
UI can show/will show arbitrary blob content or are we talking about README and License?

Reply:
@wy65701436 Thanks, it's a good catch, we should check the size in the backend to avoid parse big blob, will make the follow-up PR to fix this issue.

Reply:
Currently only README and LICENSE can be displayed in the UI, other files only display the file tree, open or download other files needs to be another proposal to discuss in the future, because based on the current API design of harbor, it's hard to open or download other files.